components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -157,6 +181,20 @@ void ViewCounterService::Shutdown() {
 }
 
 void ViewCounterService::OnUpdated(NTPBackgroundImagesData* data) {
+  if (data != GetCurrentWallpaperData())
```

Comment:
Do we need this check here?
This is only valid if `data` and GetCurrentWallpaperData() can give different data.
However, service only have one data for NTP BI.

For NTP SI/SR, this check is valid because each SI and SR use different data.